hera_cal/nucal.py

```diff
@@ -1013,3 +1032,516 @@ def project_u_model_comps_on_spec_axis(u_model_comps, spectral_filters):
+            data giving the model the flexibility to model the spatial and spectral axes. This option fits for fewer foreground parameters 
+            in the least-squares step, but can lead to slower convergence in the gradient descent step.
+        linear_solver : str, default="lu_solve"
+            Method to use for solving the linear system of equations when fitting the foreground models. Options are
+            "lu_solve", "solve", "pinv", and "lstsq". "lu_solve" uses scipy.linalg.lu_solve to solve the linear system of
+            equations, "solve" uses np.linalg.solve. "pinv" uses np.linalg.pinv to solve the linear system of equations, and
+            "lstsq" uses np.linalg.lstsq. "lu_solve" and "solve" tend to be the faster methods, but "lstsq" and "pinv" are more
+            robust.
+        linear_tol : float, default=1e-12
+            Regularization parameter for linear least-squares fit when computing the initial estimate of the foreground model.
+        share_fg_model : bool, default=False
+            If True, the foreground model for each radially-redundant group is shared across the time axis for both the least-squares and 
+            gradient descent steps.
+        spectral_filter_half_width : float, default=20e-9
+            Fourier half-width of the spectral axis DPSS filters in units of seconds.
+        spatial_filter_half_width : float, default=1
+            Half-width of the spatial axis DPSS filters in units of wavelengths.
+        eigenval_cutoff : float, default=1e-12
+            Cutoff for the eigenvalues of the DPSS filters. Only DPSS eigenvectors with eigenvalues greater than eigenval_cutoff will be used
+            for least-squares and gradient descent steps.
+        umin : float, default=None
+            Minimum u-magnitude value to include in calbration. All u-modes with magnitudes less than
+            min_u_cut will have their weights set to 0. Can also be useful for decreasing the number 
+            for foreground eigenmodes as the number of eigenmodes is roughly proportional to (umax - umin).
+        umax : float, default=None
+            Maximum u-magnitude value to include in calbration. All u-modes with magnitudes greater than
+            max_u_cut will have their weights set to 0. Can also be useful for decreasing the number 
+            for foreground eigenmodes as the number of eigenmodes is roughly proportional to (umax - umin).
+        estimate_degeneracies : bool, default=False
+            If True, the initial estimates of the redcal degeneracies will be computed from the data using traditional 
+            abscal techniques and the initial nucal model as the sky model. If False, the amplitude degeneracies will be 
+            initialized to 1 and tip-tilt degeneracies will be initialized to 0. If the data are well-calibrated,
+            setting this option to False can improve the runtime of the calibration.
```

Comment:
My thought was that this may reduce the number of iterations performed by gradient descent, but I haven't actually shown that yet. I assumed that when we're running nucal, it will usually follow a rough abscal, which is why I set this parameter to False. If you think it's unnecessary, I can always take this feature out.